### DISCUSSION TITLE
Change Update Intervall for Homeassistant Plugin

### DISCUSSION THREAD

**Discussion by Zuendapp** [ksheumaker/homeassistant-apsystems_ecur#106] [General]
Hi :),

is it possible to reduce the update intervall below 300 seconds?
In HomeAssistant I calculate my "real power Consumption" via a Shelly 3EM with a template sensor. My "real power Consumption " is the value from Shelly-3EM  (for example -400 W, and the "real consumption" would be 100W, because of the Inverter sends 500W to my power string. So I created a template sensor, which calculates the total from Shelly3EM -400W + 500W from ECU-B.
The update intervall with 300 seconds is to slow, to let me see my real power consumption without the Solarpanel.
I hope you can understand me ;).

Thank you very much!

BR

**Comment by HAEdwin** [comment]
I understand what you mean. The moment the sun reappears from behind a cloud, you will see the power consumption decrease, but the yield of the PV will lag behind. 

Unfortunately, with all ECU models, new data only becomes available once every 5 minutes. The meter shows that the PV immediately returns power, but this data comes through late. Questioning the ECU more often therefore makes no sense and has the additional adverse effect that it is more likely to be overwhelmed and get stuck.

With my P1 meter I can also read power return and although not directly from the ECU I can monitor real time return.

**Reply by Zuendapp** [reply]
Hi @HAEdwin,

thank you very much for your explanation.
So I will buy me a Powermeter like a Shelly EM or anything else.

But is the amount of power the Solar-Inverter delivers over a day, the correct one for example in my EMA Dashboard or is there anything missing like when the sun reappears from behind a cloud?

BR

**Reply by HAEdwin** [reply]
All I know is that data is sent to EMA at the same interval (once every five minutes).

**Reply by Zuendapp** [reply]
Ok, very interesting.
Does the summary values from ECU and your PowerMeter match every day?

BR

**Reply by HAEdwin** [reply]
Nope :) Never looked into this but it doesn't look good neither does it match. The ECU shows 5,93kWh and the energy DSMR meter 3,49kWh. I've got a lot of zero values on the DSMR also. But this is something completely different I have to figure out, this is independent of ECU update interval. 
![image](https://user-images.githubusercontent.com/82239374/188938399-c4439f95-79aa-432e-a6d3-a3a152bcf6a2.png)

**Reply by Zuendapp** [reply]
Ok :), I‘m not alone with this Problem.

thank you!

**Reply by HAEdwin** [reply]
I guess we didn't take into account that production drops to zero when more energy is consumed than produced which is logical. Overall the ECU is right but due to the load on the grid at home you get less production there.

**Comment by HAEdwin** [comment]
I'll close this discussion, might any other issue arise - please open a new discussion.
